CvGameCoreDLL_Expansion2/CvPlayer.cpp

```diff
@@ -12918,6 +12918,39 @@ bool CvPlayer::canReceiveGoody(CvPlot* pPlot, GoodyTypes eGoody, CvUnit* pUnit)
 	CvAssertMsg(bResult, "Cannot find goody info.");
 	kGoodyInfo.CacheResult(kResult);
 
+	// Wishing we had lambdas right about now
+	struct BestCityFinder
+	{
+		CvCity* operator()(CvPlayer& kPlayer, const CvPlot& kPlot)
```

Comment:
Did you try writing the function definition like this?:
```c++
CvCity* operator()(const CvPlayer& kPlayer, const CvPlot& kPlot)
```
this should have resolved all of your woes without needing to convert the canReceiveGoody function from being **const**.

Reply:
No, it eventually snowballs into requiring a dozen other methods to be const